keps/sig-apps/20190226-maxunavailable-for-statefulsets.md

```diff
@@ -0,0 +1,226 @@
+---
+title: Implement maxUnavailable for StatefulSets
+authors:
+  - "@krmayankk"
+owning-sig: sig-apps
+participating-sigs:
+  - sig-apps
+reviewers:
+  - "@janetkuo"
+approvers:
+  - TBD
+editor: TBD
+creation-date: 2018-12-29
+last-updated: 2018-12-29
+status: provisional
+see-also:
+  - n/a
+replaces:
+  - n/a
+superseded-by:
+  - n/a
+---
+
+# Implement maxUnavailable in StatefulSet
+
+## Table of Contents
+
+* [Table of Contents](#table-of-contents)
+* [Summary](#summary)
+* [Motivation](#motivation)
+    * [Goals](#goals)
+    * [Non-Goals](#non-goals)
+* [Proposal](#proposal)
+    * [User Stories [optional]](#user-stories-optional)
+      * [Story 1](#story-1)
+    * [Implementation Details/Notes/Constraints [optional]](#implementation-detailsnotesconstraints-optional)
+    * [Risks and Mitigations](#risks-and-mitigations)
+* [Graduation Criteria](#graduation-criteria)
+* [Implementation History](#implementation-history)
+* [Drawbacks [optional]](#drawbacks-optional)
+* [Alternatives [optional]](#alternatives-optional)
+
+[Tools for generating]: https://github.com/ekalinin/github-markdown-toc
+
+## Summary
+
+The purpose of this enhancement is to implement maxUnavailable for StatefulSet during RollingUpdate. When a StatefulSet’s 
+`.spec.updateStrategy.type` is set to `RollingUpdate`, the StatefulSet controller will delete and recreate each Pod 
+in the StatefulSet. The updating of each Pod currently happens one at a time when `spec.podManagementPolicy` is `OrderedReady`. 
+With support for `maxUnavailable`, the updating will proceed `maxUnavailable` number of pods at a time in `OrderedReady` case
+only.
```

Comment:
Isn't podManagementPolicy orthogonal to update policy? It's a policy for scaling, not update.

Reply:
correct, removed its reference. maxUnavailale is only applicable for rolling updates. It has nothing to do with podManagementPolicy